libs/go/zmscli/service.go

```diff
@@ -69,6 +69,100 @@ func (cli Zms) ShowUpdatedService(service *zms.ServiceIdentity) (*string, error)
 	return cli.dumpByFormat(service, oldYamlConverter)
 }
 
+func (cli Zms) DeleteServiceTags(dn string, sn, tagKey string, tagValue string) (*string, error) {
+	service, err := cli.Zms.GetServiceIdentity(zms.DomainName(dn), zms.SimpleName(sn))
+	if err != nil {
+		return nil, err
+	}
+	//meta := getServiceMeta(service)
+
+	tagValueArr := make([]zms.TagCompoundValue, 0)
+	if service.Tags == nil {
+		service.Tags = map[zms.CompoundName]*zms.TagValueList{}
+	}
+
+	// except given tagValue, set the same tags map
+	if tagValue != "" && service.Tags != nil {
+		currentTagValues := service.Tags[zms.CompoundName(tagKey)]
+		if currentTagValues != nil {
+			for _, curTagValue := range currentTagValues.List {
+				if tagValue != string(curTagValue) {
+					tagValueArr = append(tagValueArr, curTagValue)
+				}
+			}
+		}
+	}
```

Comment:
if this piece of code is relevant for all tag functions - please make it generic